### PR TITLE
Custom deepcopy for LocalOperator

### PR DESCRIPTION
**Pull request by VolodyaCO**
This PR adds a custom deepcopy to avoid building the LocalOperator from scratch every time it is copied.

This PR should be merged after #919 is merged.

### CODE REVIEW

**Comment by PhilipVinc** [comment]
@VolodyaCO can you please add a note to CHANGELOG about this and the other PR?

**Review by PhilipVinc — APPROVED** [review]

**Comment by VolodyaCO** [comment]
Will do @PhilipVinc

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,13 +11,15 @@
...
 * `nk.optimizer.Adam` now supports complex parameters, and you can use `nk.optimizer.split_complex` to make optimizers process complex parameters as if they are pairs of real parameters. [#1009](https://github.com/netket/netket/pull/1009)
-
 * Chunking of `MCState.expect` and `MCState.expect_and_grad` computations is now supported, which allows to bound the memory cost in exchange of a minor increase in computation time. [#1006](https://github.com/netket/netket/pull/1006) (and discussions in [#918](https://github.com/netket/netket/pull/918) and [#830](https://github.com/netket/netket/pull/830))
+* `LocalOperator` now accepts sparse matrices as input operators [#919](https://github.com/netket/netket/pull/919)
```
Comment (PhilipVinc) [inline CHANGELOG.md:15]:
```suggestion
* {ref}`nk.operator.LocalOperator` now accepts sparse matrices as input operators [#919](https://github.com/netket/netket/pull/919)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,13 +11,15 @@
...
 
 ### Breaking Changes
 
 ### Internal Changes
 * The definitions of `MCState` and `MCMixedState` have been moved to an internal module, `nk.vqs.mc` that is hidden by default. [#954](https://github.com/netket/netket/pull/954)
+* `LocalOperator` now handles all internal operators as sparse matrices [#919](https://github.com/netket/pull/919)
```
Comment (PhilipVinc) [inline CHANGELOG.md:21]:
```suggestion
* {ref}`nk.operator.LocalOperator` now handles all internal operators as sparse matrices [#919](https://github.com/netket/pull/919)
```

**Comment by PhilipVinc** [comment]
thanks for the patience!

**Pull request merged**